### PR TITLE
Add Restream Chat

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [ ] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [ ] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [ ] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

**Comment by ChrisChinchilla** [comment]
I thought I finally figured out the regex for a live check, but then got an error `key not found: :latest` and I am not sure what that means.

Also I see you really only accept "stable" versions. I have the suspicion that this is software that will be perpetually in beta, but I can submit to alternative repos if so. The AWS bucket is protected so I can't check to see if there was ever a "stable" version.

File: Casks/restream-chat.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  sha256 "32b4dae6cdce145b1f6a43b5046636bfa2188c924d792b21337c9604d3dc1809"
+
+  url "https://s3.eu-central-1.amazonaws.com/restream-chat-client/Restream+Chat-#{version}.dmg",
+      verified: "https://s3.eu-central-1.amazonaws.com/restream-chat-client/"
+  name "restream-chat"
+  desc "Keep your streaming chats in one place"
+  homepage "https://restream.io/chat/"
+
+  livecheck do
+    url "https://s3.eu-central-1.amazonaws.com/restream-chat-client/"
```
Comment (neutric) [inline Casks/restream-chat.rb:12]: This livecheck won't work as `url` errors out with "AccessDenied". Try
```
livecheck do
  url "https://s3.eu-central-1.amazonaws.com/restream-chat-client/beta-mac.yml"
  strategy :electron_builder
end
```

File: Casks/restream-chat.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    strategy :page_match
+    regex(/Restream[._+]Chat[._-]v?(\d+(?:\.\d+)+)\.dmg/i)
+  end
+
+  app "Restream Chat.app"
+
+  zap trash: [
+    "~/Library/Application Support/Restream Chat",
+  ]
```
Comment (neutric) [inline Casks/restream-chat.rb:21]: `zap trash: "~/Library/Application Support/Restream Chat"`

File: Casks/restream-chat.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+cask "restream-chat" do
+  version "2.5.4-beta"
+  sha256 "32b4dae6cdce145b1f6a43b5046636bfa2188c924d792b21337c9604d3dc1809"
+
+  url "https://s3.eu-central-1.amazonaws.com/restream-chat-client/Restream+Chat-#{version}.dmg",
+      verified: "https://s3.eu-central-1.amazonaws.com/restream-chat-client/"
+  name "restream-chat"
```
Comment (neutric) [inline Casks/restream-chat.rb:7]:
```
name "Restream Chat"
```

**Review by neutric — COMMENTED** [review]
Thanks for your contribution to `homebrew-cask`, @ChrisChinchilla! My review is in the code. Take it with a grain of salt as I am not a maintainer.

File: Casks/restream-chat.rb (hunk 1 above)
Comment (ChrisChinchilla) [reply Casks/restream-chat.rb]: @neutric Ah interesting thanks, I don't remember seeing that option in docs, which doesn't mean it wasn't there 😅

**Pull request merged**